pkg/iam/iam.go

```diff
@@ -0,0 +1,399 @@
+	"ec2:DescribeInstanceAttribute",
+	"ec2:DescribeInstances",
+	"ec2:DescribeInternetGateways",
+	"ec2:DescribeNatGateways",
+	"ec2:DescribeNetworkAcls",
+	"ec2:DescribeNetworkInterfaces",
+	"ec2:DescribeRegions",
+	"ec2:DescribeRouteTables",
+	"ec2:DescribeSecurityGroups",
+	"ec2:DescribeSubnets",
+	"ec2:DescribeTags",
+	"ec2:DescribeVolumes",
+	"ec2:DescribeVpcAttribute",
+	"ec2:DescribeVpcClassicLink",
+	"ec2:DescribeVpcClassicLinkDnsSupport",
+	"ec2:DescribeVpcs",
+	"ec2:DetachInternetGateway",
+	"ec2:DisassociateRouteTable",
+	"ec2:ModifyNetworkInterfaceAttribute",
+	"ec2:ModifyVpcAttribute",
+	"ec2:ReleaseAddress",
+	"ec2:ReplaceRouteTableAssociation",
+	"ec2:RevokeSecurityGroupEgress",
+	"ec2:RevokeSecurityGroupIngress",
+	"ec2:RunInstances",
+	"ec2:TerminateInstances",
+	"elasticloadbalancing:AddTags",
+	"elasticloadbalancing:ApplySecurityGroupsToLoadBalancer",
+	"elasticloadbalancing:AttachLoadBalancerToSubnets",
+	"elasticloadbalancing:ConfigureHealthCheck",
+	"elasticloadbalancing:CreateLoadBalancer",
+	"elasticloadbalancing:CreateLoadBalancerListeners",
+	"elasticloadbalancing:DeleteLoadBalancer",
+	"elasticloadbalancing:DeleteLoadBalancerListeners",
+	"elasticloadbalancing:DescribeLoadBalancerAttributes",
+	"elasticloadbalancing:DescribeLoadBalancers",
+	"elasticloadbalancing:DescribeTags",
+	"elasticloadbalancing:ModifyLoadBalancerAttributes",
+	"elasticloadbalancing:RegisterInstancesWithLoadBalancer",
+	"iam:AddRoleToInstanceProfile",
+	"iam:CreateInstanceProfile",
+	"iam:CreateRole",
+	"iam:DeleteInstanceProfile",
+	"iam:DeleteRole",
+	"iam:DeleteRolePolicy",
+	"iam:GetAccountAuthorizationDetails",
+	"iam:GetInstanceProfile",
+	"iam:GetRole",
+	"iam:GetUser",
+	"iam:ListAttachedGroupPolicies",
+	"iam:ListGroupsForUser",
+	"iam:ListGroupPolicies",
+	"iam:ListInstanceProfiles",
+	"iam:ListRolePolicies",
+	"iam:PassRole",
+	"iam:PutRolePolicy",
+	"iam:RemoveRoleFromInstanceProfile",
+	"route53:ChangeResourceRecordSets",
+	"route53:ChangeTagsForResource",
+	"route53:CreateHostedZone",
+	"route53:DeleteHostedZone",
+	"route53:GetChange",
+	"route53:GetHostedZone",
+	"route53:ListHostedZones",
+	"route53:ListResourceRecordSets",
+	"route53:ListTagsForResource",
+	"route53:UpdateHostedZoneComment",
+	"s3:CreateBucket",
+	"s3:DeleteBucket",
+	"s3:DeleteObject",
+	"s3:GetAccelerateConfiguration",
+	"s3:GetBucketCORS",
+	"s3:GetBucketLocation",
+	"s3:GetBucketLogging",
+	"s3:GetBucketRequestPayment",
+	"s3:GetBucketTagging",
+	"s3:GetBucketVersioning",
+	"s3:GetBucketWebsite",
+	"s3:GetEncryptionConfiguration",
+	"s3:GetLifecycleConfiguration",
+	"s3:GetObject",
+	"s3:GetObjectTagging",
+	"s3:GetReplicationConfiguration",
+	"s3:HeadBucket",
+	"s3:ListAllMyBuckets",
+	"s3:ListBucket",
+	"s3:PutBucketAcl",
+	"s3:PutBucketTagging",
+	"s3:PutObject",
+	"s3:PutObjectAcl",
+	"s3:PutObjectTagging",
+}
+
+// GetAccountAuthDetails fetches lists of users, groups, and polices.
+func GetAccountAuthDetails(svc *iam.IAM) AccountDetails {
```

Comment:
If RunPermissionsCheck is the primary entry point, it might be worth making all these other methods start with lowercase to help clarify what an external caller should be looking for. (unless I'm mistaken an external callers might use some of these other methods as well)

Reply:
You're totally right. They can always be made uppercase later, if someone else needs to use them. No reason to export them until they're needed.